test/catalog.go

```diff
@@ -175,22 +177,27 @@ func (c *Catalog) StubFunc(status corev1.ConditionStatus, reason string) func(co
 		case *build.Build:
 			Expect(object.Status.Registered).To(Equal(status))
 			Expect(object.Status.Reason).To(ContainSubstring(reason))
+			if object.Annotations != nil && object.Annotations[build.AnnotationBuildRunDeletion] == "true" {
+				Expect(object.Finalizers[0]).To(Equal(build.BuildFinalizer))
+			}
 		}
 		return nil
 	}
 }
 
 // StubBuildRunStatus asserts Status fields on a BuildRun resource
-func (c *Catalog) StubBuildRunStatus(reason string, name *string, status corev1.ConditionStatus, buildSpec build.BuildSpec) func(context context.Context, object runtime.Object, _ ...crc.UpdateOption) error {
+func (c *Catalog) StubBuildRunStatus(reason string, name *string, status corev1.ConditionStatus, buildSample *build.Build) func(context context.Context, object runtime.Object, _ ...crc.UpdateOption) error {
 	return func(context context.Context, object runtime.Object, _ ...crc.UpdateOption) error {
 		switch object := object.(type) {
 		case *build.BuildRun:
 			Expect(object.Status.Reason).To(Equal(reason))
 			Expect(object.Status.Succeeded).To(Equal(status))
 			Expect(object.Status.LatestTaskRunRef).To(Equal(name))
 			if object.Status.BuildSpec != nil {
-				Expect(*object.Status.BuildSpec).To(Equal(buildSpec))
+				Expect(*object.Status.BuildSpec).To(Equal(buildSample.Spec))
 			}
+			Expect(object.Labels[build.LabelBuild]).To(Equal(buildSample.Name))
```

Comment:
any reasons why you put assertions regarding labels, on the function that asserts object Status?

Reply:
Can you show more.

The code you pointed is the verification for the buildRun, to make sure the label of Build is added correctly.

Reply:
@zhangtbj +1 from me to what @qu1queee said. The method `StubBuildRunStatus` describes it `asserts Status fields on a BuildRun resource`. You are adding some code that does not check the status field but a label value.

Reply:
ok, I got your point.

I just want to check the label in one method, it is ok, let me add a new stub method `StubBuildRunLabel` to just check the Labels